azure-jenkins/azuredeploy.json

```diff
@@ -18,13 +18,14 @@
     "jenkinsDnsPrefix": {
       "type": "string",
       "minLength": 1,
+      "maxLength": 11,
       "metadata": {
         "description": "Globally unique DNS Name for the Public IP used to access the Virtual Machine."
       }
     }
   },
   "variables": {
-    "resourcePrefix": "jenkins",
+    "resourcePrefix": "[parameters('jenkinsDnsPrefix')]",

```

Comment:
if we're just assigning the param value to the var - can we just remove the var?  Seems unnecessary then...